source/azure_iot_provisioning_client.c

```diff
@@ -0,0 +1,963 @@
+                                                                           azureiotTOPIC_MAX, &xMQTTTopicLength );
+        }
+
+        if( az_result_failed( xCoreResult ) )
+        {
+            prvAzureIoTProvisioningClientUpdateState( xAzureIoTProvisioningClientHandle, eAzureIoTProvisioningFailed );
+            return;
+        }
+
+        if( ( xReturn =
+                  prvAzureIoTProvisioningClientRequestPayload( xAzureIoTProvisioningClientHandle,
+                                                               xAzureIoTProvisioningClientHandle->_internal.pucProvisioningScratchBuffer +
+                                                               xMQTTTopicLength,
+                                                               xAzureIoTProvisioningClientHandle->_internal.ulProvisioningScratchBufferLength -
+                                                               ( uint32_t ) xMQTTTopicLength,
+                                                               &xMQTTPayloadLength ) ) != eAzureIoTProvisioningSuccess )
+        {
+            prvAzureIoTProvisioningClientUpdateState( xAzureIoTProvisioningClientHandle, xReturn );
+            return;
+        }
+
+        ( void ) memset( ( void * ) &xMQTTPublishInfo, 0x00, sizeof( xMQTTPublishInfo ) );
+        xMQTTPublishInfo.xQOS = eAzureIoTMQTTQoS0;
+        xMQTTPublishInfo.pcTopicName = xAzureIoTProvisioningClientHandle->_internal.pucProvisioningScratchBuffer;
+        xMQTTPublishInfo.usTopicNameLength = ( uint16_t ) xMQTTTopicLength;
+        xMQTTPublishInfo.pvPayload = xAzureIoTProvisioningClientHandle->_internal.pucProvisioningScratchBuffer + xMQTTTopicLength;
+        xMQTTPublishInfo.xPayloadLength = xMQTTPayloadLength;
+        usPublishPacketIdentifier = AzureIoTMQTT_GetPacketId( &( xAzureIoTProvisioningClientHandle->_internal.xMQTTContext ) );
+
+        if( ( xResult = AzureIoTMQTT_Publish( &( xAzureIoTProvisioningClientHandle->_internal.xMQTTContext ),
+                                              &xMQTTPublishInfo, usPublishPacketIdentifier ) ) != eAzureIoTMQTTSuccess )
+        {
+            AZLogError( ( "Failed to publish. Error=%d", xResult ) );
+            xReturn = eAzureIoTProvisioningPublishFailed;
+        }
+        else
+        {
+            xReturn = eAzureIoTProvisioningSuccess;
+        }
+
+        prvAzureIoTProvisioningClientUpdateState( xAzureIoTProvisioningClientHandle, xReturn );
+    }
+    else
+    {
+        AZLogWarn( ( "Request action called in wrong state %d\r\n",
+                     xAzureIoTProvisioningClientHandle->_internal.ulWorkflowState ) );
+    }
+}
+/*-----------------------------------------------------------*/
+
+/**
+ *
+ * Implementation of parsing response action, this action is only allowed in azureiotprovisioningWF_STATE_RESPONSE
+ *
+ * */
+static void prvAzureIoTProvisioningClientParseResponse( AzureIoTProvisioningClientHandle_t xAzureIoTProvisioningClientHandle )
```

Comment:
Another nit around length - does `xAzureIoTProvisioningClientHandle` have to be so long in general?  It's scoped local so no risk of name collisions, and by time we add "_internal" it's getting long.

Again my sense if the fact this isn't a global and the file/names & dirs have "azure" all over the place already goes a long way around making intent clear here.  I'll just float out `xProvClientHandle` maybe??

Reply:
xProvClientHandle 

---
In reply to: [602632111](https://github.com/Azure/azure-iot-middleware-freertos/pull/49#discussion_r602632111) [](ancestors = 602632111)